src/Pact/Types/Exp.hs

```diff
@@ -0,0 +1,204 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE ViewPatterns #-}
+
+-- |
+-- Module      :  Pact.Types.Exp
+-- Copyright   :  (C) 2016 Stuart Popejoy
+-- License     :  BSD-style (see the file LICENSE)
+-- Maintainer  :  Stuart Popejoy <stuart@kadena.io>
+--
+-- Exp, the output of the Pact parser, and Literal.
+--
+
+module Pact.Types.Exp
+ (
+   Parsed(..),
+   Code(..),
+   Info(..),
+   renderInfo,
+   renderParsed,
+   ModuleName(..),
+   Name(..),
+   Literal(..),
+   simpleISO8601,formatLTime,
+   TypeName(..),
+   Arg(..),aInfo,aName,aType,
+   FunType(..),ftArgs,ftReturn,
+   FunTypes,funTypes,showFunTypes,
+   PrimType(..),
+   litToPrim,
+   tyInteger,tyDecimal,tyTime,tyBool,tyString,
+   tyList,tyObject,tyValue,tyKeySet,tyTable,
+   SchemaType(..),
+   TypeVarName(..),typeVarName,
+   TypeVar(..),tvName,tvConstraint,
+   Type(..),tyFunType,tyListType,tySchema,tySchemaType,tyUser,tyVar,
+   mkTyVar,mkTyVar',mkSchemaVar,
+   isAnyTy,isVarTy,isUnconstrainedTy,canUnifyWith,
+   Exp(..),eLiteral,eAtom,eBinding,eList,eLitListType,eObject,eParsed,eQualifier,eSymbol,eType,
+   _ELiteral,_ESymbol,_EAtom,_EList,_EObject,_EBinding,
+   pattern EList',pattern ELitList,pattern ELitString,pattern EAtom'
+   ) where
+
+
+import Control.Lens (makeLenses,makePrisms)
+import Control.Applicative
+import Data.List
+import Control.Monad
+import Prelude
+import Data.Text (Text,pack,unpack)
+import Data.Aeson
+import Data.String
+import Data.Default
+import Data.Char
+import Data.Thyme
+import System.Locale
+import Data.Scientific
+import GHC.Generics (Generic)
+import Data.Decimal
+import Data.Hashable
+import Text.PrettyPrint.ANSI.Leijen hiding ((<>),(<$>))
+import Control.DeepSeq
+import qualified Data.Attoparsec.Text as AP
+import Text.Trifecta (try,ident,TokenParsing,(<?>))
+import Data.Serialize (Serialize)
+
+import Pact.Types.Util
+import Pact.Types.Parser
+import Pact.Types.Info
+import Pact.Types.Type
+
+
+newtype ModuleName = ModuleName Text
+    deriving (Eq,Ord,IsString,ToJSON,FromJSON,AsString,Hashable,Pretty)
+instance Show ModuleName where show (ModuleName s) = show s
+
+-- | A named reference from source.
+data Name =
+    QName { _nQual :: ModuleName, _nName :: Text, _nInfo :: Info } |
+    Name { _nName :: Text, _nInfo :: Info }
+         deriving (Generic)
+instance Show Name where
+    show (QName q n _) = asString' q ++ "." ++ unpack n
```

Comment:
I've started using Data.String.Conv (just a single toS function for all conversions) on Doug's recommendation.  Should we all standardize on some choice?

Reply:
That's a new one, it's not a bad idea ...